src/components/Table/Table.jsx

```diff
@@ -1,203 +1,161 @@
-
-    const {
-      tableWrapperClassNames,
-      tableClassNames,
-    } = this.getComponentClassNames()
-
-    return (
-      <ThemeProvider theme={chooseSkin(skin)}>
-        <TableWrapperUI
-          {...getValidProps(rest)}
-          className={tableWrapperClassNames}
-          ref={this.setWrapperNode}
-          containerWidth={containerWidth}
-        >
-          <Scrollable
-            fadeLeft
-            fadeRight
-            scrollLockDirection="x"
-            isScrollLocked={isScrollLocked}
-          >
-            <TableUI
-              tableWidth={tableWidth}
-              withTallRows={withTallRows}
-              className={tableClassNames}
-              ref={this.setTableNode}
-            >
-              <TableHead
-                columns={columns}
-                isLoading={isLoading}
-                sortedInfo={sortedInfo}
-              />
-
-              <TableBody
-                rows={data}
-                columns={columns}
-                isTableCollapsed={isTableCollapsed}
-                maxRowsToDisplay={maxRowsToDisplay}
-                onRowClick={onRowClick}
-              />
-            </TableUI>
-          </Scrollable>
-
-          {isLoading && <LoadingUI className={`${TABLE_CLASSNAME}__Loading`} />}
-
-          {maxRowsToDisplay && isTableCollapsed ? (
-            <Button
-              style={{ marginLeft: '14px' }}
-              kind="link"
-              className={`${TABLE_CLASSNAME}__Expander`}
-              onClick={this.handleExpanderClick}
-            >
-              {expanderText ? expanderText.collapsed : 'View All'}
-            </Button>
-          ) : null}
-
-          {maxRowsToDisplay && !isTableCollapsed ? (
-            <Button
-              style={{ marginLeft: '14px' }}
-              kind="link"
-              className={`${TABLE_CLASSNAME}__Expander`}
-              onClick={this.handleExpanderClick}
-            >
-              {expanderText ? expanderText.expanded : 'Collapse'}
-            </Button>
-          ) : null}
-        </TableWrapperUI>
-      </ThemeProvider>
-    )
-  }
-}
-Table.defaultProps = {
-  columns: [],
-  data: [],
-  'data-cy': 'Table',
-  skin: defaultSkin,
-  tableWidth: { min: '700px' },
-  containerWidth: '100%',
-  sortedInfo: {
+export function Table({
+  className,
+  columns = [],
+  containerWidth = '100%',
+  data = [],
+  'data-cy': dataCy = 'Table',
+  expanderText,
+  isLoading = false,
+  isScrollLocked = true,
+  maxRowsToDisplay,
+  onExpand = noop,
+  onSelectRow = noop,
+  onRowClick = null,
+  selectKey = 'id',
+  skin = defaultSkin,
+  sortedInfo = {
     columnKey: null,
     order: null,
   },
-  isLoading: false,
-  isScrollLocked: true,
-  onRowClick: null,
-  wrapperRef: noop,
-  tableRef: noop,
-  onExpand: noop,
-  withTallRows: false,
+  tableClassName,
+  tableWidth = { min: '700px' },
+  withSelectableRows = false,
+  withTallRows = false,
+}) {
+  const [state, dispatch] = useReducer(reducer, {
```

Comment:
I think even though it is for the initialState of the reducer, the getDisplayTableData will be called on each component render phase (I could be wrong here)

Reply:
Good catch, memoized it and it fixes that